controls/views.py

```diff
@@ -2400,3 +2402,114 @@ def poam_export(request, system_id, format='xlsx'):
+                project.save()
+                messages.add_message(request, messages.INFO, f'Created a new project with id: {project.id}.')
+
+            system_id = project.system.id
+            # Retrieve identified System
+            system = System.objects.get(id=system_id)
+            #Import questionnaire data
+            log_output = []
+            try:
+                from collections import OrderedDict
+                data = json.loads(imported_jsondata, object_pairs_hook=OrderedDict)
+            except Exception as e:
+                log_output.append("There was an error reading the export file.")
+            else:
+                try:
+                    # Update project data.
+                    project.import_json(data, request.user, "imp", lambda x: log_output.append(x))
+                except Exception as e:
+                    log_output.append(str(e))
+
+            # Log output
+            logger.info(
+                event="project JSON import",
+                object={"object": "project", "id": project.id, "title": project.title, "log_output": log_output},
+                user={"id": request.user.id, "username": request.user.username}
+            )
+
+            # Load and get the components then dump
+            for k, val in enumerate(json.loads(imported_jsondata).get('component-definitions')):
+                oscal_component_json = json.dumps(json.loads(imported_jsondata).get('component-definitions')[k])
+                result = ComponentImporter().import_component_as_json(oscal_component_json, request)
+
+        return HttpResponseRedirect("/systems/{}/controls/selected".format(system_id))
+
+def project_export(request, project_id):
+    """
+    Export an entire project's components and control content
+    """
+    # Of the project in the current system. pick one project to export
+    project = Project.objects.get(id=project_id)
+    system_id = project.system.id
+    # Retrieve identified System
+    system = System.objects.get(id=system_id)
+
+    # Retrieve related selected controls if user has permission on system
+    if request.user.has_perm('view_system', system):
+
+        # Iterate through the elements associated with the system get all statements produced for each
+        oscal_comps = []
+        for element in system.producer_elements:
+            # Implementation statement OSCAL JSON
+            impl_smts = element.statements_produced.filter(consumer_element=system.root_element)
+            component = OSCALComponentSerializer(element, impl_smts).as_json()
+            oscal_comps.append(component)
+
+    # TODO: multiple export types
+
+    questionnaire_data = json.dumps(project.export_json(include_metadata=True, include_file_content=True))
+    data = json.loads(questionnaire_data)
+    for oscal_comp in oscal_comps:
+        data['component-definitions'] = json.loads(oscal_comp)
+    data['component-definitions'] = [json.loads(oscal_comp) for oscal_comp in oscal_comps]
```

Comment:
These seem like duplicates of each other. Are both really needed?

Reply:
true got right of the first one.